components/datadog/apps/nginx/ecsFargate.go

```diff
@@ -0,0 +1,159 @@
+	appName := "nginx-fg"
+	namer := e.Namer.WithPrefix(appName)
+	opts = append(opts, e.WithProviders(config.ProviderAWS, config.ProviderAWSX))
+
+	EcsFargateComponent := &EcsFargateComponent{}
+	if err := e.Ctx.RegisterComponentResource("dd:apps", namer.ResourceName("grp"), EcsFargateComponent, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(EcsFargateComponent))
+
+	alb, err := lb.NewApplicationLoadBalancer(e.Ctx, namer.ResourceName("lb"), &lb.ApplicationLoadBalancerArgs{
+		Name:           e.CommonNamer.DisplayName(32, pulumi.String(appName)),
+		SubnetIds:      e.RandomSubnets(),
+		Internal:       pulumi.BoolPtr(true),
+		SecurityGroups: pulumi.ToStringArray(e.DefaultSecurityGroups()),
+		DefaultTargetGroup: &lb.TargetGroupArgs{
+			Name:       e.CommonNamer.DisplayName(32, pulumi.String(appName)),
+			Port:       pulumi.IntPtr(80),
+			Protocol:   pulumi.StringPtr("HTTP"),
+			TargetType: pulumi.StringPtr("ip"),
+		},
+		Listener: &lb.ListenerArgs{
+			Port:     pulumi.IntPtr(80),
+			Protocol: pulumi.StringPtr("HTTP"),
+		},
+	}, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	serverContainer := &ecs.TaskDefinitionContainerDefinitionArgs{
+		Name:  e.CommonNamer.DisplayName(255, pulumi.String("server")),
+		Image: pulumi.String("ghcr.io/datadog/apps-nginx-server:main"),
+		DockerLabels: pulumi.StringMap{
+			"com.datadoghq.ad.checks": pulumi.String(utils.JSONMustMarshal(
+				map[string]interface{}{
+					"nginx": map[string]interface{}{
+						"init_config": map[string]interface{}{},
+						"instances": []map[string]interface{}{
+							{
+								"nginx_status_url": "http://%%host%%/nginx_status",
+							},
+						},
+					},
+				},
+			)),
+			"com.datadoghq.ad.tags": pulumi.String("[\"ecs_task_type:fargate\"]"),
+		},
+		Cpu:       pulumi.IntPtr(100),
+		Memory:    pulumi.IntPtr(96),
+		Essential: pulumi.BoolPtr(true),
+		DependsOn: ecs.TaskDefinitionContainerDependencyArray{
+			ecs.TaskDefinitionContainerDependencyArgs{
+				ContainerName: pulumi.String("datadog-agent"),
+				Condition:     pulumi.String("HEALTHY"),
+			},
+		},
+		PortMappings: ecs.TaskDefinitionPortMappingArray{
+			ecs.TaskDefinitionPortMappingArgs{
+				ContainerPort: pulumi.IntPtr(80),
+				HostPort:      pulumi.IntPtr(80),
+				Protocol:      pulumi.StringPtr("tcp"),
+			},
+		},
+		HealthCheck: ecs.TaskDefinitionHealthCheckArgs{
+			Command: pulumi.StringArray{
+				pulumi.String("CMD-SHELL"),
+				pulumi.String("apk add curl && curl --fail http://localhost || exit 1"),
+			},
+		},
+		LogConfiguration: ecsClient.GetFirelensLogConfiguration(pulumi.String("nginx"), pulumi.String("nginx"), apiKeySSMParamName),
+	}
+
+	serverTaskDef, err := ecsClient.FargateTaskDefinitionWithAgent(e, e.CommonNamer.ResourceName("nginx-server"), e.CommonNamer.DisplayName(255, pulumi.String("server")), 1024, 2048, map[string]ecs.TaskDefinitionContainerDefinitionArgs{"server": *serverContainer}, apiKeySSMParamName, fakeintake, opts...)
```

Comment:
It’s not needed to use a `namer` for the `name` and `family` parameters of `ecsClient.FargateTaskDefinitionWithAgent` because this function already uses a namer:
https://github.com/DataDog/test-infra-definitions/blob/40f6fda565a098ae6e4111bd95db84fd22952aac/resources/aws/ecs/fargateService.go#L42
https://github.com/DataDog/test-infra-definitions/blob/40f6fda565a098ae6e4111bd95db84fd22952aac/resources/aws/ecs/fargateService.go#L52

Reply:
Got it, I'll wait on #513